core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1069,14 +1073,32 @@ public void loadIntoContext(ShardSearchRequest request, SearchContext context, Q
     }
 
     /**
+     * Fetch {@linkplain FieldStats} for a field. These stats are cached until the shard changes.
+     * @param shard the shard to use with the cache key
+     * @param searcher searcher to use to lookup the field stats
+     * @param field the actual field
+     * @param fieldType the field's type
+     */
+    public FieldStats<?> getFieldStats(IndexShard shard, Engine.Searcher searcher, String field, MappedFieldType fieldType)
```

Comment:
good!! can we also move the `MappedFieldType` lookup in there since we have the `shard` reference anyway so we can safe one parameter?